emmet-core/emmet/core/qc_tasks.py

```diff
@@ -0,0 +1,616 @@
+        description="Input molecule and calc details for the QChem calculation",
+    )
+
+    prev_rem_params: Optional[Dict[str, Any]] = Field(
+        None,
+        description="Parameters from a previous qchem calculation in the series",
+    )
+
+    rem: Dict[str, Any] = Field(
+        None,
+        description="Parameters from the rem section of the current QChem calculation",
+    )
+
+    level_of_theory: Optional[Union[str, LevelOfTheory]] = Field(
+        None, description="Level of theory used in the qchem calculation"
+    )
+
+    task_type: Optional[Union[str, TaskType]] = Field(
+        None,
+        description="The type of the QChem calculation : optimization, single point ... etc.",
+    )
+
+    tags: Union[List[str], None] = Field(
+        [], title="tag", description="Metadata tagged to a given task."
+    )
+
+    solvation_lot_info: Optional[Union[Dict[str, Any], str]] = Field(
+        None,
+        description="Str or Dict representation of the solvent method used for the calculation",
+    )
+
+    special_run_type: Optional[str] = Field(
+        None, description="Special workflow name (if applicable)"
+    )
+
+    smiles: Optional[str] = Field(
+        None,
+        description="Simplified molecular-input line-entry system (SMILES) string for the molecule involved "
+        "in this calculation.",
+    )
+
+    calc_type: Optional[Union[str, CalcType]] = Field(
+        None,
+        description="A combined dictionary representation of the task type along with the level of theory used",
+    )
+
+    @classmethod
+    def from_qchem_calc_doc(cls, calc_doc: Calculation) -> "InputDoc":
+        """
+        Create qchem calculation input summary from a qchem calculation document.
+
+        Parameters
+        ----------
+        calc_doc
+            A QChem calculation document.
+
+        Returns
+        --------
+        InputDoc
+            A summary of the input molecule and corresponding calculation parameters
+        """
+        # TODO : modify this to get the different variables from the task doc.
+        # print(calc_doc)
+        return cls(
+            initial_molecule=calc_doc.input.initial_molecule,
+            rem=calc_doc.input.rem,
+            level_of_theory=calc_doc.level_of_theory.value,
+            task_type=calc_doc.task_type.value,
+            tags=calc_doc.input.tags,
+            solvation_lot_info=calc_doc.solvation_lot_info,
+            # special_run_type = calc_doc.input.special_run_type,
+            # smiles = calc_doc.input.smiles,
+            calc_type=calc_doc.calc_type.value,
+        )
+
+
+class CustodianDoc(BaseModel):
+    corrections: Optional[List[Any]] = Field(
+        None,
+        title="Custodian Corrections",
+        description="List of custodian correction data for calculation.",
+    )
+
+    job: Optional[Dict[str, Any]] = Field(
+        None,
+        title="Custodian Job Data",
+        description="Job data logged by custodian.",
+    )
```

Comment:
Is there an equivalent in VASP? Hopefully a lot of boilerplate you could borrow if so